src/validators/uuid.rs

```diff
@@ -0,0 +1,184 @@
+            Version::UUIDv3 => 3,
+            Version::UUIDv4 => 4,
+            Version::UUIDv5 => 5,
+        }
+    }
+}
+
+impl From<u8> for Version {
+    fn from(u: u8) -> Self {
+        match u {
+            1 => Version::UUIDv1,
+            3 => Version::UUIDv3,
+            4 => Version::UUIDv4,
+            5 => Version::UUIDv5,
+            _ => unreachable!(),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct UuidValidator {
+    strict: bool,
+    version: Option<Version>,
+}
+
+impl BuildValidator for UuidValidator {
+    const EXPECTED_TYPE: &'static str = "uuid";
+
+    fn build(
+        schema: &PyDict,
+        config: Option<&PyDict>,
+        _definitions: &mut DefinitionsBuilder<CombinedValidator>,
+    ) -> PyResult<CombinedValidator> {
+        let py = schema.py();
+        let version = schema.get_as::<u8>(intern!(py, "version"))?.map(Version::from);
+        Ok(Self {
+            strict: is_strict(schema, config)?,
+            version,
+        }
+        .into())
+    }
+}
+
+impl Validator for UuidValidator {
+    fn validate<'s, 'data>(
+        &'s self,
+        py: Python<'data>,
+        input: &'data impl Input<'data>,
+        extra: &Extra,
+        _definitions: &'data Definitions<CombinedValidator>,
+        _recursion_guard: &'s mut RecursionGuard,
+    ) -> ValResult<'data, PyObject> {
+        let class = get_uuid_type(py)?;
+        if let Some(py_input) = input.input_is_instance(class) {
+            Ok(py_input.to_object(py))
+        } else if extra.strict.unwrap_or(self.strict) && input.is_python() {
+            Err(ValError::new(
+                ErrorType::UuidExactType {
+                    class_name: self.get_name().to_string(),
+                },
+                input,
+            ))
+        } else {
+            let dc = create_class(class)?;
+            let uuid = self.get_uuid(py, input)?;
+            self.set_dict_call(py, dc.as_ref(py), &uuid)?;
+            Ok(dc)
+        }
+    }
+
+    fn different_strict_behavior(
+        &self,
+        _definitions: Option<&DefinitionsBuilder<CombinedValidator>>,
+        ultra_strict: bool,
+    ) -> bool {
+        !ultra_strict
+    }
+
+    fn get_name(&self) -> &str {
+        Self::EXPECTED_TYPE
+    }
+
+    fn complete(&mut self, _definitions: &DefinitionsBuilder<CombinedValidator>) -> PyResult<()> {
+        Ok(())
+    }
+}
+
+impl UuidValidator {
+    fn get_uuid<'s, 'data>(&'s self, py: Python<'data>, input: &'data impl Input<'data>) -> ValResult<'data, Uuid> {
+        let uuid = match input.validate_str(false) {
+            Ok(either_string) => {
+                let cow = either_string.as_cow()?;
+                let uuid_str = cow.as_ref();
+                match Uuid::parse_str(uuid_str) {
+                    Ok(uuid) => uuid,
+                    Err(e) => return Err(ValError::new(ErrorType::UuidParsing { error: e.to_string() }, input)),
+                }
+            }
```

Comment:
I think we probably need to add a branch for `bytes`?

Reply:
Done, WDYT?